### PR TITLE
Add BLEManagerImpl for mbed-os

### LINKED ISSUE #27: Implement BLEManager
#### Problem
The BLEManager is required for proper BLE communication.

 #### Proposed Solution
Implement it. Mbed OS stub is available [here](https://github.com/ARMmbed/connectedhomeip/blob/d17d9becd3bf63ad72251cb51f96294994848bbc/src/platform/ESP32/BLEManagerImpl.h#L76-L88)

### LINKED ISSUE #28: Implement BLEPlatformManagerDelegate
#### Problem
Ble communication requires implementation of the BLEPlatformManagerDelegate.

 #### Proposed Solution
Implement it, Mbed stub is [here](https://github.com/ARMmbed/connectedhomeip/blob/d17d9becd3bf63ad72251cb51f96294994848bbc/src/platform/ESP32/BLEManagerImpl.h#L92-L105)

### LINKED ISSUE #29: Implement BLE application delegate.
#### Problem
There is no Mbed implementation of that class.

 #### Proposed Solution
Implement it with Mbed BLE. Stub is [here](https://github.com/ARMmbed/connectedhomeip/blob/d17d9becd3bf63ad72251cb51f96294994848bbc/src/platform/ESP32/BLEManagerImpl.h#L109)

### PR DESCRIPTION
**Pull request by fkjagodzinski**
#### Problem

Missing implementation of BLE layer that is split into following classes:
- `BLEManager`,
- `BlePlatformDelegate`,
- `BleApplicationDelegate`.

 #### Summary of Changes

Add a `BLEManagerImpl` class that provides all the missing methods and does the GAP and GATT event handling.

Fixes #27,
Fixes #28,
Fixes #29,

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,23 @@
 # Connected Home over IP
```
Comment (fkjagodzinski) [inline README.md:1]: Sorry about that. This file will be removed from this PR.

**Review by pan- — COMMENTED** [review]
Thanks @fkjagodzinski . Ping me if you need any help to complete the implementation

File: src/platform/mbed/BLEManagerImpl.cpp [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    mServiceMode = ConnectivityManager::kCHIPoBLEServiceMode_Enabled;
+    mFlags       = CHIP_DEVICE_CONFIG_CHIPOBLE_ENABLE_ADVERTISING_AUTOSTART ? kFlag_AdvertisingEnabled : 0;
+    mGAPConns    = 0;
+    // memset(mSubscribedConns, 0, sizeof(mSubscribedConns));
+
+    ble::BLE & mBLEInterface = ble::BLE::Instance();
+    mBLEInterface.onEventsToProcess(ScheduleBLEProcessing);
+
+    // FIXME Use CHIP event handling?
+    osStatus rc = mEventQueueThread.start(mbed::callback(&mEventQueue, &events::EventQueue::dispatch_forever));
```
[the remark this answers is not in the document]
Comment (fkjagodzinski) [reply src/platform/mbed/BLEManagerImpl.cpp]: Done.

File: src/platform/mbed/BLEManagerImpl.cpp [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    VerifyOrExit(rc == osOK, err = CHIP_ERROR_INTERNAL);
+    mBLEInterface.init(HandleInintComplete);
+
+#if _IMPL_READY_BLEPLATFORMDELEGATE && _IMPL_READY_BLEAPPLICATIONDELEGATE
```
[the remark this answers is not in the document]
Comment (fkjagodzinski) [reply src/platform/mbed/BLEManagerImpl.cpp]: Yeah, updated as suggested.

File: src/platform/mbed/BLEManagerImpl.cpp [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+{
+    mEventQueue.call(mbed::callback(&(context->ble), &ble::BLE::processEvents));
+}
+
+void BLEManagerImpl::HandleInitComplete(ble::BLE::InitializationCompleteCallbackContext * params)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    VerifyOrExit(params->error != BLE_ERROR_NONE, err = CHIP_ERROR_INTERNAL);
+
```
[the remark this answers is not in the document]
Comment (fkjagodzinski) [reply src/platform/mbed/BLEManagerImpl.cpp:146]: Done.

File: src/platform/mbed/BLEManagerImpl.h [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+}
+
+inline BLEManager::CHIPoBLEServiceMode BLEManagerImpl::_GetCHIPoBLEServiceMode(void)
+{
+    return mServiceMode;
+}
+
+inline bool BLEManagerImpl::_IsAdvertisingEnabled(void)
+{
+    return GetFlag(mFlags, kFlag_AdvertisingEnabled);
```
[the remark this answers is not in the document]
Comment (fkjagodzinski) [reply src/platform/mbed/BLEManagerImpl.h:173]: Yes, pretty much. `kFlag_Advertising` means the advertising is currently active. But there are some cases when `kFlag_Advertising` is false even though advertising has been enabled, e.g. when `CHIP_DEVICE_CONFIG_CHIPOBLE_SINGLE_CONNECTION` is used and we already have an active connection.

File: src/platform/mbed/BLEManagerImpl.cpp [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+using namespace ::chip::System;
+
+namespace chip {
+namespace DeviceLayer {
+namespace Internal {
+
+/*
+namespace {
+
+const bt_uuid_128 UUID128_CHIPoBLEChar_RX =
```
[the remark this answers is not in the document]
Comment (fkjagodzinski) [reply src/platform/mbed/BLEManagerImpl.cpp]: Done

**Comment by fkjagodzinski** [comment]
This is finally ready for review. Additional clean-up will come with updates requested here.

**Review by pan- — APPROVED** [review]
Looks good to me. I have to test it again before merging though.

**Pull request merged**